src/org/omegat/MainClassLoader.java

```diff
@@ -0,0 +1,103 @@
+/**************************************************************************
+ OmegaT - Computer Assisted Translation (CAT) tool
+          with fuzzy matching, translation memory, keyword search,
+          glossaries, and translation leveraging into updated projects.
+
+ Copyright (C) 2021 Hiroshi Miura
+               2018 Mordechai Meisels (Original licensed Apache-2.0)
+               Home page: http://www.omegat.org/
+               Support center: https://omegat.org/support
+
+ This file is part of OmegaT.
+
+ OmegaT is free software: you can redistribute it and/or modify
+ it under the terms of the GNU General Public License as published by
+ the Free Software Foundation, either version 3 of the License, or
+ (at your option) any later version.
+
+ OmegaT is distributed in the hope that it will be useful,
+ but WITHOUT ANY WARRANTY; without even the implied warranty of
+ MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ GNU General Public License for more details.
+
+ You should have received a copy of the GNU General Public License
+ along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ **************************************************************************/
+
+package org.omegat;
+
+import java.io.File;
+import java.io.IOException;
+import java.net.MalformedURLException;
+import java.net.URL;
+import java.net.URLClassLoader;
+import java.nio.file.Paths;
+
+/**
+ * @author Hiroshi Miura
+ */
+public final class MainClassLoader extends URLClassLoader {
```

Comment:
Sorry, one more question about this class: Can you explain why we need it?

What does this class loader do that the default class loaders don't?

Reply:
I should say that I don't know very much about how class loaders work. Sorry!

Reply:
Because swing cannot find an external LaF of plugins. It causes ClassNotFound error (or NPE in Swing component) without it.

It is because Swing has a limitation, that use SystemClassLoader to find UI component even when child class loader load LaF and register the loader to UIManager. (It seems feature bug but not documented the feature).

Reply:
Now I found a way without specify custom system loader for JVM argument.

Reply:
From another point of view. System's default class loader don't allow adding  new URL(plugin) to search path, no method to do it. URLClassLoader class has a addURL(URL url) method that is **protected**.

https://github.com/openjdk/jdk/blob/6765f902505fbdd02f25b599f942437cd805cad1/src/java.base/share/classes/java/net/URLClassLoader.java#L389

MainClassLoader has methods, `add(URL url)` and `addJarToClassPath(String jarName)` and its scope is **default** (package local).
Only a Main class can manipulate a class search path in same security reason.

MainClassLoader#add method give a chance to loading plugins without restart in future.

Reply:
> Now I found a way **without** specifying a custom system loader for JVM argument, but we still need custom MainClassLoader that has a capability to add URLs of plugins after class loader created.

I tried running without the JVM argument and I got these errors:

```
34572: Error: java.lang.NullPointerException 
34572: Error:   at java.util.concurrent.ConcurrentHashMap.putVal(ConcurrentHashMap.java:1011) 
34572: Error:   at java.util.concurrent.ConcurrentHashMap.putIfAbsent(ConcurrentHashMap.java:1535) 
34572: Error:   at java.lang.ClassLoader.getClassLoadingLock(ClassLoader.java:457) 
34572: Error:   at java.lang.ClassLoader.loadClass(ClassLoader.java:398) 
34572: Error:   at java.lang.ClassLoader.loadClass(ClassLoader.java:351) 
34572: Error:   at org.omegat.util.gui.UIDesignManager.setLookAndFeel(UIDesignManager.java:138) 
34572: Error:   at org.omegat.util.gui.UIDesignManager.setDefaultTheme(UIDesignManager.java:118) 
34572: Error:   at org.omegat.Main.runGUI(Main.java:288) 
34572: Error:   at org.omegat.Main.main(Main.java:184) 
34572: Error: java.lang.NullPointerException 
34572: Error:   at org.omegat.util.gui.UIDesignManager.initialize(UIDesignManager.java:189) 
34572: Error:   at org.omegat.core.Core.initializeGUI(Core.java:237) 
34572: Error:   at org.omegat.Main.runGUI(Main.java:295) 
34572: Error:   at org.omegat.Main.main(Main.java:184) 
```

The last error was unhandled so it crashed the process.

I think this will cause a lot of support issues if the JVM argument really becomes required. I think a lot of people run OmegaT by double-clicking OmegaT.jar, at least during troubleshooting.

Reply:
Actually I get those errors even if I do use the JVM args.

I invoked as

```sh
java -Djava.system.class.loader=org.omegat.MainClassLoader -jar build/install/OmegaT/OmegaT.jar
```

Reply:
When calling `UIManager.setLookAndFeel(String name)`, it always use **SystemClassLoader**. When we want to load class that was loaded by our own class loader, it went NPE or ClassNotFound.
So I try always to use `UIManager.setLookAndFeel(LookAndFeel object)` form.
```
        Class<?> clazz = classLoader.loadClass(getThemeClassName(laf));
        UIManager.setLookAndFeel((LookAndFeel) clazz.getConstructor().newInstance());
```

When investigation of  call stack, it was not  handled  by our custom `MainClassLoader`, but `SystemClassLoader`.

```
34572: Error:   at java.lang.ClassLoader.loadClass(ClassLoader.java:351) 
34572: Error:   at org.omegat.util.gui.UIDesignManager.setLookAndFeel(UIDesignManager.java:138) 
```

Reply:
I'd like to reproduce the issue. I've tested with OpenJDK 11 on Linux, so I'd like to test with sevral JDK versions.

Reply:
I can reproduce the issue.
It happens when no theme configuration in omegat.prefs.
Current theme key is something like 
`   <theme>javax.swing.UIManager$LookAndFeelInfo[Default com.sun.java.swing.plaf.gtk.GTKLookAndFeel]</theme>`

but loading default is looking `Default` then it was not found in set of Laf information.
 
The fix is to load the default class and override configuration with proper value.